src/components/user/notification.tsx

```diff
@@ -124,141 +125,174 @@ export function Notification({
+      }
+      if (keys.indexOf(part) > -1) {
+        return <React.Fragment key={index}>{replaceables[part]}</React.Fragment>
+      }
+      return part
+    })
+  }
+
   function renderText() {
+    const actor = renderUser(event.actor)
+
     switch (event.__typename) {
       case 'SetThreadStateNotificationEvent':
-        return (
-          <>
-            {renderUser(event.actor)} hat einen {renderThread(event.thread.id)}{' '}
-            {event.archived ? 'archiviert' : 'unarchiviert'}
-          </>
+        return parseString(
+          event.archived
+            ? '%actor% hat einen %thread% archiviert.'
+            : '%actor% hat einen %thread% unarchiviert.',
+          {
+            thread: renderThread(event.thread.id),
+          }
         )
+
       case 'CreateCommentNotificationEvent':
-        return (
-          <>
-            {renderUser(event.actor)} hat einen{' '}
-            <StyledLink href={`/${event.comment.id}`}>Kommentar</StyledLink> in{' '}
-            einem {renderThread(event.thread.id)} erstellt.
-          </>
+        return parseString(
+          '%actor% hat einen %comment% in einem %thread% erstellt.',
+          {
+            thread: renderThread(event.thread.id),
+            comment: (
+              <StyledLink href={`/${event.comment.id}`}>Kommentar</StyledLink>
+            ),
+          }
         )
+
       case 'CreateThreadNotificationEvent':
-        return (
-          <>
-            {renderUser(event.actor)} hat einen {renderThread(event.thread.id)}{' '}
-            in {renderObject(event.object)} erstellt.
-          </>
+        return parseString(
+          '%actor% hat einen %thread% in einem %object% erstellt.',
+          {
+            thread: renderThread(event.thread.id),
+            object: renderObject(event.object),
+          }
         )
+
       case 'CreateEntityNotificationEvent':
-        return (
-          <>
-            {renderUser(event.actor)} hat {renderObject(event.entity)} erstellt.
-          </>
-        )
+        return parseString('%actor% hat %object% erstellt.', {
+          object: renderObject(event.entity),
+        })
+
       case 'SetLicenseNotificationEvent':
-        return (
-          <>
-            {renderUser(event.actor)} hat die Lizenz von{' '}
-            {renderObject(event.repository)} geändert.
-          </>
+        return parseString(
+          '%actor% hat die Lizenz von %repository% geändert.',
+          {
+            repository: renderObject(event.repository),
+          }
         )
+
       case 'CreateEntityLinkNotificationEvent':
-        return (
-          <>
-            {renderUser(event.actor)} hat {renderObject(event.child)} mit{' '}
-            {renderObject(event.parent)} verknüpft.
-          </>
-        )
+        return parseString('%actor% hat %child% mit %parent% verknüpft.', {
+          child: renderObject(event.child),
+          parent: renderObject(event.parent),
+        })
+
```

Comment:
this is the parser interface currently. 
a bit verbose (duplicate code for e.g. `child: renderObject(event.child)`) but clean and typescript is happy. 
should I move this logic into the parseString function? (possibly annoying because TS looses the relation to type, but less code overall)

Reply:
ping @inyono